File: src/ApplicationCore/WordDictionary/Word.cs
namespace ApplicationCore.WordDictionary;
public class Word
{
    public string? Name { set; get; }
    public string? Pronounciation { get; set; }          // British
    public string? PronounciationAm { get; set; }        // American
    public int Frequency { get; set; }
    public string? MeaningShort { get; set; }
    public string? MeaningLong { get; set; }
    public string? SoundUrl { get; set; }
    public string? ExampleSoundUrl { get; set; }
    public string? SimilarWords { get; set; }
    public int Id { get; set; }

    public Word(string name)
    {
        Name = name;
    }
}

File: src/webapp-angular/src/app/domain/model/word.ts
export class Word {
  name!: string;
  pronounciation: string | undefined;          // British
  pronounciationAm: string | undefined;        // American
  frequency: number | undefined;
  similarWords: string | undefined;
  meaningShort: string | undefined;
  meaningLong: string | undefined;
  meaningOther: string | undefined;            // meaning in other language
  soundUrl: string | undefined;
  exampleSoundUrl: string | undefined;
  id: number | undefined;

  // fields for viewing
  viewTime: Date | undefined;
  easiness: number | undefined;                // user choose the easiness:: not started yet,  -2, -1:hard, 0:normal, 1:easy ,2 
  viewInterval: number | undefined;            // viewTime + viewInterval = next view time
  totalViewed: number | undefined;
  startTime: Date | undefined;
}

export class WordStudyModel {
  id?: number;
  userName?: string | null;
  wordName?: string | null;
  startTimeUtc?: Date;
  lastStudyTimeUtc?: Date;
  studyCount?: number;
  isClosed?: boolean;
  daysToStudy?: number; // lastStudyTimeUtc + daysToStudy = nextStudyTimeUtc to remind user to study
  word?: Word | null;
}
